ntoskrnl/kd/kdmain.c

```diff
@@ -153,11 +153,39 @@ KdpEnterDebuggerException(IN PKTRAP_FRAME TrapFrame,
 #ifdef KDBG
         else if (ExceptionCommand == BREAKPOINT_LOAD_SYMBOLS)
         {
+            PKD_SYMBOLS_INFO UnsafeSymbolsInfo;
+            KD_SYMBOLS_INFO SymbolsInfo;
             PLDR_DATA_TABLE_ENTRY LdrEntry;
 
-            /* Load symbols. Currently implemented only for KDBG! */
-            if(KdbpSymFindModule(((PKD_SYMBOLS_INFO)ExceptionRecord->ExceptionInformation[2])->BaseOfDll, NULL, -1, &LdrEntry))
-                KdbSymProcessSymbols(LdrEntry);
+            UnsafeSymbolsInfo = (PKD_SYMBOLS_INFO)ExceptionRecord->ExceptionInformation[2];
+            if (PreviousMode != KernelMode)
+            {
+                _SEH2_TRY
+                {
+                    ProbeForRead(UnsafeSymbolsInfo,
+                                 sizeof(*UnsafeSymbolsInfo),
+                                 1);
+                    RtlCopyMemory(&SymbolsInfo,
+                                  UnsafeSymbolsInfo,
+                                  sizeof(SymbolsInfo));
+                }
+                _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+                {
+                    UnsafeSymbolsInfo = NULL;
+                }
+                _SEH2_END;
+            }
+            else
+            {
+                SymbolsInfo = *UnsafeSymbolsInfo;
```

Comment:
Just for my education: Why does this copy not use RtlCopyMemory, but inside the seh block the copy is done with RtlCopyMemory?

Reply:
The RtlCopyMemory was a spontaneous change to make sure the assignment doesn't get optimized out. It's probably not necessary, but in either case it certainly should be consistent between the two cases.